docs/modules/SDDs/pages/0026-commodore-component-testing.adoc

```diff
@@ -0,0 +1,66 @@
+= SDD 0025 - Commodore Component Instantiation
+
+:sdd_author:    Christian Cremer
+:sdd_owner:     Christian Cremer
+:sdd_reviewers: SIG Syn
+:sdd_date:      2020-11-06
+:sdd_status:    draft
+include::partial$meta-info-table.adoc[]
+
+[NOTE]
+.Summary
+====
+This SDD describes possible options to test Commodore components locally and in CI.
+====
+
+== Motivation
+
+Currently Commodore components don't support compiling and testing isolated within a component itself, one needs a whole catalog.
+Testing a component with different parameters is crucial in ensuring component quality.
+
+=== Goals
+
+* Define how we want to test Commodore components.
+
+=== Non-Goals
+
+* Dictate which kind of tests a component author or contributor must provide in order for contributions to be accepted.
+* Dictate test "coverage" rules.
+
+== Design Proposal
+
+Component templates will provide a scaffold for an easy getting started with testing.
+Components are to be tested with Go unit tests and OpenPolicyAgent's Conftest policies using the Rego language.
+The advantage of using Go unit tests allows component authors to deserialize rendered YAML back into actual Kubernetes structs, allowing easy assertions on expected values.
+The advantage of Writing policies using the Rego syntax allows ensuring certain standards over a multiple of manifests (for example define well-known common labels).
+Combining both test methods should cover a wide range of test cases.
+It should be easy for component authors to run the tests on their local workstation, without needing a Commodore catalog compililation.
+A library for Go unit tests to define common helper functions should reduce the need to write boilerplate code for testing.
+A central set of policies tailored for Syn can be applied against all components in order to maintain a unified "good practice" across the components.
```

Comment:
I would allow usage of other testing tools on a components author discretion. For https://github.com/stackrox/kube-linter (https://www.civo.com/learn/yaml-best-practices-using-kubelinter) looks also promising.

Reply:
I have added something in the non-goals, is that also ok for you?